deps/rabbit/src/rabbit_fifo.erl

```diff
@@ -1501,14 +1518,27 @@ return_all(Meta, #?MODULE{consumers = Cons} = State0, Effects0, ConsumerId,
                 end, {State, Effects0}, Checked).
 
 %% checkout new messages to consumers
-checkout(#{index := Index} = Meta, OldState, State0, Effects0) ->
+checkout(#{index := Index} = Meta, #?MODULE{cfg = #cfg{resource = QName}} = OldState, State0,
+         Effects0) ->
     {State1, _Result, Effects1} = checkout0(Meta, checkout_one(Meta, State0),
                                             Effects0, {#{}, #{}}),
     case evaluate_limit(Index, false, OldState, State1, Effects1) of
         {State, true, Effects} ->
-            update_smallest_raft_index(Index, State, Effects);
+            case have_active_consumers_changed(OldState, State) of
```

Comment:
I am not sure about adding more code this very hot path. is there nowhere else we an do this and achieve the same?

Reply:
We have to detect changes in the active consumers, unless we keep track of them on a different way for this exclusive use I'm not sure how we can do it

Reply:
If we don't do this here, what would be the observable effects? If it's a minor delay in consumer topology changes, we can live with that over more code executed on the hot path.

Reply:
I've removed the notifications for enqueue/return/credit/settle/discard operations. It seems enough for federation purposes (our only decorator?) to only notify on checkout/node up/node down/config updates. At least, federation test suite still passes.